tests/test_auto_provisioning.py

```
@@ -0,0 +1,72 @@
+    # Use dummy credentials - we're only testing DB provisioning, not authentication
+    broker = AngelOne(
+        api_key="DUMMY_KEY",
+        client_code="DUMMY_CODE",
+        password="DUMMY_PIN",
+        totp_key="DUMMY_TOTP"
+    )
+    
+    # 3. Verify DB was created
+    if db_path.exists():
+        file_size = db_path.stat().st_size
+        print(f"✅ SUCCESS! Instruments DB created automatically")
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)